test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -810,61 +800,53 @@ TEST_F(AmetsuchiTest, FindTxByHashTest) {
   ASSERT_TRUE(storage);
   auto blocks = storage->getBlockQuery();
 
-  iroha::pubkey_t pubkey1, pubkey2;
-  pubkey1.at(0) = 1;
-  pubkey2.at(0) = 2;
-
-  CreateRole createRole;
-  createRole.role_name = "user";
-  createRole.permissions = {can_add_peer, can_create_asset, can_get_my_account};
-
-  Transaction tx1;
-  tx1.creator_account_id = "admin1";
-
-  tx1.commands.push_back(std::make_shared<CreateRole>(createRole));
-  CreateDomain createDomain;
-  createDomain.domain_id = "domain";
-  createDomain.user_default_role = "user";
-  tx1.commands.push_back(std::make_shared<CreateDomain>(createDomain));
-
-  CreateAccount createAccount;
-  createAccount.account_name = "user1";
-  createAccount.domain_id = "domain";
-  createAccount.pubkey = pubkey1;
-  tx1.commands.push_back(std::make_shared<CreateAccount>(createAccount));
-
-  CreateRole createRole2;
-  createRole2.role_name = "user2";
-  createRole2.permissions = {
-      can_add_peer, can_create_asset, can_get_my_account};
-
-  Transaction tx2;
-  tx2.commands.push_back(std::make_shared<CreateRole>(createRole2));
-  CreateDomain createDomain2;
-  createDomain2.domain_id = "domain2";
-  createDomain2.user_default_role = "user";
-  tx2.commands.push_back(std::make_shared<CreateDomain>(createDomain2));
-
-  Block block;
-  block.transactions.push_back(tx1);
-  block.transactions.push_back(tx2);
-  block.height = 1;
-  block.prev_hash.fill(0);
-  block.txs_number = block.transactions.size();
-  block.hash = iroha::hash(block);
+  shared_model::crypto::PublicKey pubkey1(std::string("1", 32));
+  shared_model::crypto::PublicKey pubkey2(std::string("2", 32));
+
+  auto txn1 =
+      TestTransactionBuilder()
+          .creatorAccountId("admin1")
+          .createRole("user",
+                      std::set<std::string>{
+                          can_add_peer, can_create_asset, can_get_my_account})
+          .createDomain("domain", "user")
+          .createAccount("user1", "domain", pubkey1)
+          .build();
+
+  auto txn2 =
+      TestTransactionBuilder()
+          .creatorAccountId("admin1")
+          .createRole("user2",
+                      std::set<std::string>{
+                          can_add_peer, can_create_asset, can_get_my_account})
+          .createDomain("domain2", "user")
+          .build();
+
+  auto block = TestBlockBuilder()
+                   .transactions(std::vector<shared_model::proto::Transaction>(
+                       {txn1, txn2}))
+                   .height(1)
+                   .prevHash(shared_model::crypto::Hash(std::string("0", 32)))
+                   .txNumber(2)
+                   .build();
 
   apply(storage, block);
 
   // TODO: 31.10.2017 luckychess move tx3hash case into a separate test after
   // ametsuchi_test redesign
-  auto tx1hash = iroha::hash(tx1).to_string();
-  auto tx2hash = iroha::hash(tx2).to_string();
+  auto tx1hash =
+      std::string{txn1.hash().blob().begin(), txn1.hash().blob().end()};
```

Comment:
`BlockQuery` interface can use `shared_model` types to avoid such constructions.